os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -121,10 +146,16 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 #endif /* CONFIG_MM_ASSERT_ON_FAIL */
 
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
+#if !defined(CONFIG_WATCHDOG) || defined(CONFIG_WATCHDOG) && defined(CONFIG_MM_ASSERT_ON_FAIL)
+	/* This can print logs for more than 3 minutes, which can be rebooted by watchdog.
+	 * To do this, there is an action to stop the watchdog, but it is only when CONFIG_MM_ASSERT_ON_FAIL
+	 * is enabled. Otherwise don't print heapinfo_parse_heap() log.
+	 */
 	for (int idx = startidx; idx <= endidx; idx++) {
 		heapinfo_parse_heap(&heap[idx], HEAPINFO_DETAIL_ALL, HEAPINFO_PID_ALL);
```

Comment:
heapinfo_parse_heap can be called from the application also. Even in that case, we can have a watchdog reset. 
So, how about we move the watchdog stop logic inside the heapinfo_parse_heap() ?

```
heapinfo_parse_heap
{
    # if watchdog
         pause watchdogs
    # endif

    print logs


    # if watchdog
        resume watchdog
    # endif
}
```

Reply:
yes Thank you, I think this is good idea, 
but, if can stop the watch, we should know able to reboot. (if force stop watchdog, we must reboot.)
because forced stop or pause of the watchdog breaks the time guaranteed by the watchdog.
if logic is in heapinfo_parse_heap. we don't know able to reboot 

I think pause and resume is not right
Thank you for your review~

Reply:
> heapinfo_parse_heap can be called from the application also. Even in that case, we can have a watchdog reset.

Isn't the watchdog reset here happening because of disabling irq in mm_manage_alloc_fail? I think it might not be needed in this case.